manifests/ssl.pp

```diff
@@ -2,10 +2,24 @@
 #
 # Apache resources specific for SSL
 #
-class apache::ssl {
+class apache::ssl (
+  $ssl_port            = params_lookup( 'ssl_port' ),
+  $ssl_source          = params_lookup( 'ssl_source' ),
```

Comment:
I wasn't sure, but it didn't seem to make sense to have a node declaration like:

class { 'apache':
  ssl_template => 'site/apache/ssl.conf.erb',
}
class { 'apache::ssl': }

I was concerned about it causing confusion when 1) user specifies the param in the apache class and doesn't declare apache:ssl, 2) user attempts to specify the param in the apache::ssl class, and 3) when using an ENC, where the param would be specified as apache_ssl_template might make it look like apache::ssl::template instead of apache::ssl_template if that makes sense.